pkg/eventprocessor/eventfeed/impl/eventfeed.go

```diff
@@ -146,6 +152,8 @@ func (ef *EventFeed) Start(
 				break
 			}
 
+			ef.sm.SetLastSeenBlockNumber(ef.chainID, toHeight)
```

Comment:
This is the place where we're setting the last seen block number. It's right after we received a new block notification and we adjusted to account for reorgs. So, it's not exactly the last seen. it's the last seen minus our configuration. For Ethereum `mainnet`, it's the last seen minus 5.